### PR TITLE
refactor: update transfer to use runtime.Environment

### PR DESCRIPTION
**Pull request by bznein**
## Description

Opened to trigger e2e

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/callbacks/testing/simapp/app.go [hunk 1]
```diff
@@ -459,7 +459,9 @@ func NewSimApp(
 	// since fee middleware will wrap the IBCKeeper for underlying application.
 	// NOTE: the Transfer Keeper's ICS4Wrapper can later be replaced.
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
-		appCodec, runtime.NewKVStoreService(keys[ibctransfertypes.StoreKey]), app.GetSubspace(ibctransfertypes.ModuleName),
+		appCodec,
+		runtime.NewEnvironment(runtime.NewKVStoreService(keys[ibctransfertypes.StoreKey]), logger.With(log.ModuleKey, "x/transfer")),
```
Comment (bznein) [inline modules/apps/callbacks/testing/simapp/app.go:463]: Here, and elsewhere, not sure if `x/transfer` is right. Just followed what happened in other PRs

Comment (damiannolan) [reply modules/apps/callbacks/testing/simapp/app.go:463]: I think "x/ibc-transfer" should be good

**Comment by bznein** [comment]
@damiannolan this is ready for review. I think the only failing E2Es are the same failing on the base PR

File: testing/simapp/app.go [hunk 2]
```diff
@@ -541,7 +541,8 @@ func NewSimApp(
 	// since fee middleware will wrap the IBCKeeper for underlying application.
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
 		appCodec,
-		runtime.NewKVStoreService(keys[ibctransfertypes.StoreKey]), app.GetSubspace(ibctransfertypes.ModuleName),
+		runtime.NewEnvironment(runtime.NewKVStoreService(keys[ibctransfertypes.StoreKey]), logger.With(log.ModuleKey, "x/trasfer")),
```
Comment (DimitrisJim) [inline testing/simapp/app.go:544]:
```suggestion
		runtime.NewEnvironment(runtime.NewKVStoreService(keys[ibctransfertypes.StoreKey]), logger.With(log.ModuleKey, "x/ibc-trasfer")),
```

Though don't we have a constant for this module key field passed in?

**Review by DimitrisJim — APPROVED** [review]
Trusting CI 🙏 Thanks very much @bznein!

File: testing/simapp/app.go (hunk 2 above)
Comment (damiannolan) [reply testing/simapp/app.go:544]: trasfer -> transfer :D 

another good reason to use constants

Comment (DimitrisJim) [reply testing/simapp/app.go:544]: Fucking lol

File: modules/apps/transfer/keeper/keeper.go [hunk 3]
```diff
@@ -23,13 +22,12 @@ import (
...
 	porttypes "github.com/cosmos/ibc-go/v9/modules/core/05-port/types"
 	host "github.com/cosmos/ibc-go/v9/modules/core/24-host"
-	"github.com/cosmos/ibc-go/v9/modules/core/exported"
 	coretypes "github.com/cosmos/ibc-go/v9/modules/core/types"
 )
 
 // Keeper defines the IBC fungible transfer keeper
 type Keeper struct {
-	storeService   corestore.KVStoreService
+	appmodule.Environment
```
Comment (DimitrisJim) [inline modules/apps/transfer/keeper/keeper.go:30]: additional minor nit (noticed while looking at other PR!)

```suggestion
	appmodule.Environment

```

**Comment by gjermundgaraba** [comment]
@damiannolan I guess you can take over this one?

File: testing/simapp/app.go [hunk 4]
```diff
@@ -95,7 +95,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/server/config"
 	servertypes "github.com/cosmos/cosmos-sdk/server/types"
 	"github.com/cosmos/cosmos-sdk/std"
-	"github.com/cosmos/cosmos-sdk/testutil/testdata/testpb"
+	testdata_pulsar "github.com/cosmos/cosmos-sdk/testutil/testdata/testpb"
```
Comment (DimitrisJim) [inline testing/simapp/app.go:98]: what a weird ass linting error though

Comment (damiannolan) [reply testing/simapp/app.go:98]: Yeah, underscores are verboten aren't they? I fixed this in another branch, must have slipped through from merge conflicts

File: modules/core/02-client/types/height_test.go
Comment (damiannolan) [inline modules/core/02-client/types/height_test.go:1]: revert

File: modules/core/keeper/msg_server_test.go
Comment (damiannolan) [inline modules/core/keeper/msg_server_test.go:1]: revert

**Review by damiannolan — APPROVED** [review]
I'll do events and authz in a follow up PR

**Pull request merged**